### PR TITLE
[vcpkg_configure_make] Report warning when unresolved arguments are passed in

### LINKED ISSUE #22535: vcpkg_configure_make should warn/error on unparsed arguments
```
vcpkg_configure_make(
   SOURCE_PATH "${SOURCE_PATH}"
   OPTIOS
      --test=34
)
```
Works without a warning or error. It should print a warning or stops with an error because `OPTIOS` is an unknown argument.  
Discovered in https://github.com/microsoft/vcpkg/pull/22421#issuecomment-1012959754

### PR DESCRIPTION
**Pull request by JackBoosY**
In https://github.com/microsoft/vcpkg/pull/22321, I wrote a typo keyword `OPTIOS` and it was not recoginzed or report warnign message, that's not good.

Fix that.

Fixes #22535.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Depends on #22534

File: scripts/cmake/vcpkg_configure_make.cmake [hunk 1]
```diff
@@ -228,11 +228,15 @@ endmacro()
 function(vcpkg_configure_make)
     # parse parameters such that semicolons in options arguments to COMMAND don't get erased
     cmake_parse_arguments(PARSE_ARGV 0 arg
-        "AUTOCONFIG;SKIP_CONFIGURE;COPY_SOURCE;DISABLE_VERBOSE_FLAGS;NO_ADDITIONAL_PATHS;ADD_BIN_TO_PATH;USE_WRAPPERS;DETERMINE_BUILD_TRIPLET"
+        "AUTOCONFIG;SKIP_CONFIGURE;COPY_SOURCE;DISABLE_VERBOSE_FLAGS;NO_ADDITIONAL_PATHS;ADD_BIN_TO_PATH;NO_DEBUG;USE_WRAPPERS;DETERMINE_BUILD_TRIPLET"
```
Comment (JackBoosY) [inline scripts/cmake/vcpkg_configure_make.cmake:231]: `NO_DEBUG` is used below but is not declare here.

Comment (dg0yt) [reply scripts/cmake/vcpkg_configure_make.cmake:231]: And it is not documented.

**Review by LilyWangLL — APPROVED** [review]

**Comment by JackBoosY** [comment]
```
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:85 (file):
  file failed to open for reading (No such file or directory):

    /mnt/vcpkg-ci/buildtrees/liburing/build-x64-linux-dbg-out.log
Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:187 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/liburing/portfile.cmake:20 (vcpkg_install_make)
  scripts/ports.cmake:142 (include)
```
No idea why it happened.

**Comment by JackBoosY** [comment]
`vcpkg_configure_make` does not match `vcpkg_build_make` when `NO_DEBUG` is enabled.

File: scripts/cmake/vcpkg_configure_make.cmake (hunk 1 above)
Comment (Neumann-A) [reply scripts/cmake/vcpkg_configure_make.cmake:231]: is NO_DEBUG used anywhere ? Couldn't the same be achieved by setting `VCPKG_BUILD_TYPE` before the command?

Comment (JackBoosY) [reply scripts/cmake/vcpkg_configure_make.cmake:231]: @Neumann-A See https://github.com/microsoft/vcpkg/blob/f54a7b00130e54a398e9c852d200616b670e991c/ports/liburing/portfile.cmake#L13-L17

Comment (JackBoosY) [reply scripts/cmake/vcpkg_configure_make.cmake:231]: Yeah I suddenly realized that this option has no effect now unless the makefile doesn't support debug mode.

**Pull request merged**